daemon/containerd/image_tag.go

```diff
@@ -60,7 +60,22 @@ func (i *ImageService) TagImageWithReference(ctx context.Context, imageID image.
 			return errdefs.System(err)
 		}
 	}
-
 	logger.Info("image created")
+
+	if err == nil {
+		// The tag succeeded, check if the source image is dangling
+		sourceDanglingImg, err := is.Get(ctx, danglingImageName(target.Digest))
+
+		if err == nil {
```

Comment:
Also wondering if it'd be more idiomatic to do an early return instead;

```go
	// The tag succeeded, check if the source image is dangling
	sourceDanglingImg, err := is.Get(ctx, danglingImageName(target.Digest))
	if err != nil {
		if !cerrdefs.IsNotFound(err) {
			logger.WithError(err).Warn("unexpected error when checking if source image is dangling")
		}
		return nil
	}

	// Delete the source dangling image, as it's no longer dangling.
	if err := is.Delete(ctx, sourceDanglingImg.Name); err != nil {
		logger.WithError(err).Warn("unexpected error when deleting dangling image")
	}

	return nil
}
```

Reply:
Changed, thanks